themes/inbeat/layouts/partials/site/ad-mockup-generator/generalMixin.html

```diff
@@ -6,6 +6,10 @@
           },
         methods: {
             downloadPng(target,name){
+
+              if (this.showUnlockPopup()) {
+                  return;
+              }
```

Comment:
So this mixin depends on the other mixin? Feels misleading and easy to forget.

Reply:
Yea, we don't want to trigger the "download" when popup is shown (Popup return true if is visible) - Really not the best way to do it, but we needed a quick integration. Also this popup is a temporary feature that will likely be updated or remove later

Reply:
I understand the goal. It's just weird to have mixins depending on each other, since they should add their own behaviour. If I just add the `generalMixin` and not the other one, this function will break. If they are dependant on each other, they should possibly be together?

Let's face it, my code wasn't clean in the beginning anyways so I don't really mind haha just a suggestion.